src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -41,22 +88,132 @@ public MtAmazonDynamoDbStreamsBySharedTable(AmazonDynamoDBStreams amazonDynamoDb
+            if (unionSize > limit) {
+                break;
+            }
+            result = nextResult;
+            List<Record> union = new ArrayList<>(unionSize);
+            union.addAll(mtResult.getRecords());
+            union.addAll(nextMtResult.getRecords());
+            mtResult = nextMtResult.withRecords(union);
+        }
+
+        return mtResult;
+    }
+
+    private GetRecordsResult toMtResult(Function<Map<String, AttributeValue>, FieldValue> fieldValueFunction,
+                                        GetRecordsResult result) {
+        Function<Record, MtRecord> toMtRecord = record -> toMtRecord(fieldValueFunction, record);
+        return new GetRecordsResult()
+                .withNextShardIterator(result.getNextShardIterator())
+                .withRecords(result.getRecords().stream()
+                        .map(toMtRecord)
+                        .filter(this::matchesContext)
+                        .collect(toList()));
+    }
+
+    private MtRecord toMtRecord(Function<Map<String, AttributeValue>, FieldValue> fieldValueFunction,
+                                Record record) {
+        FieldValue fieldValue = fieldValueFunction.apply(record.getDynamodb().getKeys());
+        MtAmazonDynamoDbContextProvider mtContext = mtDynamoDb.getMtContext();
+        // execute in record tenant context to get table mapping
+        TableMapping tableMapping = mtContext.withContext(fieldValue.getMtContext(),
+                mtDynamoDb::getTableMapping, fieldValue.getTableIndex());
+        ItemMapper itemMapper = tableMapping.getItemMapper();
+        StreamRecord streamRecord = record.getDynamodb();
+        return new MtRecord()
+                .withAwsRegion(record.getAwsRegion())
+                .withEventID(record.getEventID())
+                .withEventName(record.getEventName())
+                .withEventSource(record.getEventSource())
+                .withEventVersion(record.getEventVersion())
+                .withContext(fieldValue.getMtContext())
+                .withTableName(fieldValue.getTableIndex())
+                .withDynamodb(new StreamRecord()
+                        .withKeys(itemMapper.reverse(streamRecord.getKeys()))
+                        .withNewImage(itemMapper.reverse(streamRecord.getNewImage()))
+                        .withOldImage(itemMapper.reverse(streamRecord.getOldImage()))
+                        .withSequenceNumber(streamRecord.getSequenceNumber())
+                        .withStreamViewType(streamRecord.getStreamViewType())
+                        .withApproximateCreationDateTime(streamRecord.getApproximateCreationDateTime())
+                        .withSizeBytes(streamRecord.getSizeBytes()));
+    }
+
+    private boolean matchesContext(MtRecord mtRecord) {
+        String context = mtDynamoDb.getMtContext().getContext();
+        return context == null || context.equals(mtRecord.getContext());
```

Comment:
I see, you're treating no context as "all" contexts.  This puts more responsibility on higher level clients to make sure there is a context provided where appropriate.

Reply:
Yes. Will change this as discussed.